add geolocation samples (#944)
GoogleChrome/chrome-extensions-samples: functional-samples/cookbook.geolocation-offscreen/README.md

This recipe shows how to get `geolocation` access within a service worker

## Context

> In extensions, geolocation uses the same web APIs you are used to. Instead, if you want to get geolocation information, use the same [`navigator.geolocation`](geolocation) DOM API that any web site normally would. This demo shows how to access this data in a Service Worker (via [offscreenDocument][2]. We also have samples for a [popup][3], or a [content script][4]

Reviewer: ```suggestion
To get geolocation information in extensions, use the same [`navigator.geolocation`][6] DOM API that any website normally would. This demo shows how to access this data in a Service Worker (via [offscreenDocument][2]. We also have samples for a [popup][3], or a [content script][4].
```

## Running this extension

1. Clone this repository.

> 1. Load this directory in Chrome as an [unpacked extension][2].

Reviewer: ```suggestion
1. Load this directory in Chrome as an [unpacked extension][1].
```

1. Open the Extension menu and click the extension named "Geolocation - offscreen".

[1]: https://developer.chrome.com/docs/extensions/mv3/getstarted/development-basics/#load-unpacked
[2]: https://developer.chrome.com/docs/extensions/reference/offscreen/
[3]: functional-samples/cookbook.geolocation-popup

> [4]: functional-samples/cookbook.geolocation-contentscript

Reviewer: ```suggestion
[4]: functional-samples/cookbook.geolocation-contentscript
[6]: https://developer.mozilla.org/docs/Web/API/Navigator/geolocation
```

This recipe shows how to get `geolocation` access within a service worker

## Context

To get geolocation information in extensions, use the same [`navigator.geolocation`][6] DOM API that any website normally would. This demo shows how to access this data in a Service Worker (via [offscreenDocument][2]. We also have samples for a [popup][3], or a [content script][4].

> ## Running this extension

Reviewer: I tested the offscreen docs sample. When I click on the action icon when I'm on the chrome://extensions page, nothing happens. I inspected the SW and I didn't see any logs 

Next, I tried navigating to example.com and clicking on the action icon. I inspected the service worker, but there was no log.

We should probably add more steps.

This recipe shows how to get `geolocation` access within a service worker

## Context

> To get geolocation information in extensions, use the same [`navigator.geolocation`][6] DOM API that any website normally would. This demo shows how to access this data in a Service Worker (via [offscreenDocument][2]. We also have samples for a [popup][3], or a [content script][4].

Author: sorry, which point is being dittoed?

> To get geolocation information in extensions, use the same [`navigator.geolocation`][6] DOM API that any website normally would. This demo shows how to access this data in a Service Worker (via [offscreenDocument][2]. We also have samples for a [popup][3], or a [content script][4].

Reviewer 2: Oops, sorry - just the formatting changes here: https://github.com/GoogleChrome/chrome-extensions-samples/pull/944#discussion_r1231148022

> To get geolocation information in extensions, use the same [`navigator.geolocation`][6] DOM API that any website normally would. This demo shows how to access this data in a Service Worker (via [offscreenDocument][2]. We also have samples for a [popup][3], or a [content script][4].

Author: happy to update - is it "offscreen document", and not "Offscreen Document"?

> This recipe shows how to get `geolocation` access within a service worker

Reviewer 2: Nit: Can we move this down between the header and "Overview"?

Author: fixed with [1dbac70](https://github.com/GoogleChrome/chrome-extensions-samples/pull/944/commits/1dbac7034801f93237ddfa43f3a9a44f9f7552eb)

# geolocation via offscreen document

## Overview

> To get geolocation information in extensions, use the same [`navigator.geolocation`][6] DOM API that any website normally would. This demo shows how to access this data in a Service Worker (via [offscreen document][2]. We also have samples for a [popup][3], or a [content script][4].

Reviewer 2: ```suggestion
To get geolocation information in extensions, use the same [`navigator.geolocation`][6] DOM API that any website normally would. This demo shows how to access this data in a Service Worker (via [offscreen document][2]). We also have samples for a [popup][3], or a [content script][4].
```

Author: fixed with [1dbac70](https://github.com/GoogleChrome/chrome-extensions-samples/pull/944/commits/1dbac7034801f93237ddfa43f3a9a44f9f7552eb)
